lib/moab/file_signature.rb

```diff
@@ -139,24 +158,17 @@ def hash
       @size.to_i
     end
 
-    # @api internal
+    # @deprecated
+    # this method is a holdover from an earlier version.  use the class method .from_file going forward.
+    # @api external
     # @param pathname [Pathname] The location of the file to be digested
     # @return [FileSignature] Generate a FileSignature instance containing size and checksums for a physical file
     def signature_from_file(pathname)
-      @size = pathname.size
-      md5_digest = Digest::MD5.new
-      sha1_digest = Digest::SHA1.new
-      sha256_digest = Digest::SHA2.new(256)
-      pathname.open("r") do |stream|
-        while (buffer = stream.read(8192))
-          md5_digest.update(buffer)
-          sha1_digest.update(buffer)
-          sha256_digest.update(buffer)
-        end
-      end
-      @md5 = md5_digest.hexdigest
-      @sha1 = sha1_digest.hexdigest
-      @sha256 = sha256_digest.hexdigest
+      file_signature = self.class.from_file(pathname)
+      @size = file_signature.size
```

Comment:
if these are attributes, should it be self.size???

Reply:
The old instance method is very nearly a constructor.  It sets all 4 attributes and returns the object `self`.  That means that the returned object has basically no relationship to the previous instance, suggesting it should be returning a new object, which is why we deprecated this method and made the class method `self.from_file`.  

For backwards-compatibility, we maintain the method signature and behavior of the old method, but since we stole its guts up in `from_file`, to avoid code duplication, `signature_from_file` now actually does create a new object and takes the attributes from it.  This ensures that we don't have any logical divergence between the two.

Also, more simply, `@size = self.size` is a tautology.

Reply:
I corrected a bunch of those in my separate PR such that if it were merged and this one rebased, I'd expect it to fall out.  The instance variable assignment is obviously from the original implementation.

Reply:
@ndushay incorporating those changes here.